bin/ofs/src/fuse.rs

```diff
@@ -636,6 +637,107 @@ impl PathFilesystem for Fuse {
             entries: relative_paths.chain(children).skip(offset as usize).boxed(),
         })
     }
+    async fn rename2(
+        &self,
+        req: Request,
+        origin_parent: &OsStr,
+        origin_name: &OsStr,
+        parent: &OsStr,
+        name: &OsStr,
+        _flags: u32,
+    ) -> Result<()> {
+        log::debug!(
+            "rename2(origin_parent={:?}, origin_name={:?}, parent={:?}, name={:?})",
+            origin_parent,
+            origin_name,
+            parent,
+            name
+        );
+        self.rename(req, origin_parent, origin_name, parent, name)
+            .await
+    }
+
+    async fn lseek(
+        &self,
+        _req: Request,
+        path: Option<&OsStr>,
+        fh: u64,
+        offset: u64,
+        whence: u32,
+    ) -> Result<ReplyLSeek> {
+        log::debug!(
+            "lseek(path={:?}, fh={}, offset={}, whence={})",
+            path,
+            fh,
+            offset,
+            whence
+        );
+
+        let file = self.get_opened_file(FileKey::try_from(fh)?, path)?;
+
+        if !file.is_read && !file.is_write {
+            Err(Errno::from(libc::EACCES))?;
+        }
+
+        let whence = whence as i32;
+
+        let offset = if whence == libc::SEEK_CUR || whence == libc::SEEK_SET {
+            offset
+        } else if whence == libc::SEEK_END {
+            let metadata = self
+                .op
+                .stat(&path.unwrap().to_string_lossy())
+                .await
+                .map_err(opendal_error2errno)?;
+            let content_size = metadata.content_length();
+
+            if content_size >= offset as _ {
+                content_size as u64 - offset
+            } else {
+                0
+            }
+        } else {
+            return Err(libc::EINVAL.into());
+        };
+
+        Ok(ReplyLSeek { offset })
+    }
+
+    async fn copy_file_range(
```

Comment:
I feel like it's incorrect to support `copy_file_range` since we didn't do zero copy here.

Reply:
I think it's Okay for the use case, `copy_file_range` is for reducing the file system context switch between kernel mode and user mode. That fits in the use case in `ofs`.